stdlib/LinearAlgebra/src/lu.jl

```diff
@@ -240,8 +240,15 @@ end
 # We can't assume an ordered field so we first try without pivoting
 function lu(A::AbstractMatrix{T}; check::Bool = true) where T
     S = lutype(T)
+    onex = one(S)
     AA = similar(A, S)
     copyto!(AA, A)
+    try
+        if applicable(float, onex) && float(onex) === onex
+            return lu!(AA, Val(true); check = check)
+        end
+    catch
+    end
```

Comment:
this will also swallow exceptions thrown by `lu!`, which is probably not what we want?

Maybe do something like:
```jl
isfloat = try; applicable(float, onex) && float(onex) === onex; catch; false; end
isfloat && return lu!(AA, Val(true); check = check)
```